File: Gifs_Watcher/app/src/main/java/com/example/gifs_watcher/managers/ApiManager.java
package com.example.phonenetworks.managers;

import com.example.phonenetworks.services.NetworkService;

import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class ApiManager {

    final static String BASE_URL = "https://public.opendatasoft.com/api/records/1.0/";
    private NetworkService networkService = null;
    private static ApiManager instance;

    public NetworkService getNetworkService() {
        return networkService;
    }

    public static ApiManager getInstance() {
        if (instance == null) {
            instance = new ApiManager();
        }

        return instance;
    }

    private ApiManager() {
        createRetrofitNetWork();
    }

    private void createRetrofitNetWork() {

        HttpLoggingInterceptor interceptor = new HttpLoggingInterceptor();
        interceptor.level(HttpLoggingInterceptor.Level.BODY);
        OkHttpClient client = new OkHttpClient.Builder().addInterceptor(interceptor).build();


        Retrofit retrofitClock = new Retrofit.Builder()
                .baseUrl(BASE_URL)
                .client(client)
                .addConverterFactory(GsonConverterFactory.create())
                .build();

        this.networkService = retrofitClock.create(NetworkService.class);
    }


}

File: Gifs_Watcher/app/src/main/java/com/example/gifs_watcher/datasource/ApiManager.kt
package com.example.gifs_watcher.datasource

import com.example.gifs_watcher.utils.services.TenorService
import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object ApiManager {
    val BASE_URL = "https://g.tenor.com/"
    private var tenorService: TenorService? = null

    init {
        createRetrofit()
    }
    fun getTenorService(): TenorService? {
        return tenorService
    }

    private fun createRetrofit() {

        val interceptor = HttpLoggingInterceptor()
        interceptor.level = HttpLoggingInterceptor.Level.BODY

        val client: OkHttpClient = OkHttpClient.Builder().addInterceptor(interceptor).build()

        val retrofit = Retrofit.Builder()
            .baseUrl(BASE_URL)
            .client(client)
            .addConverterFactory(GsonConverterFactory.create())
            .build()

        tenorService = retrofit.create(TenorService::class.java)
    }
}